src/resources/resource-manager.js

```diff
@@ -1,13 +1,30 @@
 import i18next from 'i18next';
 import icu from 'i18next-icu';
+import cloneDeep from '../utils/clone-deep.js';
 import { initReactI18next } from 'react-i18next';
 import { SUPPORTED_UI_LANGUAGES } from './ui-language.js';
 
 const DEFAULT_NAMESPACE = 'default';
 
 class ResourceManager {
-  constructor(resourceBundles) {
-    this._resourceBundles = resourceBundles;
+  constructor(...resourceBundles) {
+    this._resourceBundles = this._mergeResourceBundles(resourceBundles);
+  }
+
+  _mergeResourceBundles(bundles) {
+    const result = [];
+    for (const bundle of bundles) {
+      for (const entry of bundle) {
```

Comment:
I think you could do `bundles.flat().forEach(bundle => { .. })`, or?
And small thing, could `existing` become `existingBundle` please?

Reply:
Yup, good idea, both done (although I named it `existingEntry`)!